### PR TITLE
Enable Prometheus scraping using mTLS

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**

Due to [this issue](https://github.com/kyma-project/kyma/issues/11179), we want to have a setup in which Prometheus runs outside of the Istio-Service-Mesh, but it should be possible that Prometheus scrapes the metrics using mTLS. 

Thus, we came up with the following solution:
- Deploy Prometheus with Istio sidecar-proxy. &#8594; SDS certificates are rotated and written to a volume that is shared with Prometheus.
- Sidecar of Prometheus does not intercept requests for Prometheus. &#8594; Prometheus does not have an active part in the service mesh. We have to do it in this way because, Prometheus’s model of direct endpoint access is incompatible with Istio’s sidecar proxy model.
- `PeerAuthentication` of Kyma components can be converted back to `STRICT`. Since, `STRICT` is the default value, the corresponding PeerAuthentication files are removed.
- The `ServiceMonitors` of each Kyma component needs the mounted TLS certificates for authentication.
- `insecureSkipVerify`  is set to true for each `ServiceMonitor` since Prometheus does not support `Istio secure naming`

**Related issue(s)**

#11179

### CODE REVIEW

File: resources/monitoring/templates/prometheus-operator/servicemonitor.yaml [hunk 1]
```diff
@@ -10,17 +10,10 @@ metadata:
 spec:
   endpoints:
   {{- if .Values.prometheusOperator.tls.enabled }}
-  - port: https
```
Comment (a-thaler) [inline resources/monitoring/templates/prometheus-operator/servicemonitor.yaml:13]: why to remove it, just assure that the enabled setting is false by default

File: resources/monitoring/values.yaml [hunk 2]
```diff
@@ -685,7 +689,15 @@ grafana:
 ##
 kubeApiServer:
   enabled: true
-  tlsConfig:
+  scheme: https
+
+    ## tlsConfig: TLS configuration to use when scraping the endpoint. For example if using istio mTLS.
+    ## Of type: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#tlsconfig
+  tlsConfig:
```
Comment (a-thaler) [inline resources/monitoring/values.yaml:703]: Does this really work? Have you checked the apiserver metrics?

File: resources/monitoring/charts/kube-state-metrics/values.yaml [hunk 3]
```diff
@@ -94,7 +94,7 @@ tolerations: []
 
 # Annotations to be added to the pod
 podAnnotations:
-  sidecar.istio.io/inject: "false"
```
Comment (a-thaler) [inline resources/monitoring/charts/kube-state-metrics/values.yaml]: Please do not add a sidecar to kube-state-metrics

File: components/event-publisher-proxy/config/event-publisher-nats/120-peerauthentication.yaml [hunk 4]
```diff
@@ -10,4 +10,4 @@ spec:
       app.kubernetes.io/instance: eventing
   mtls:
     mode: "PERMISSIVE"
----
```
Comment (a-thaler) [inline components/event-publisher-proxy/config/event-publisher-nats/120-peerauthentication.yaml:13]: Is there a good reason to keep it, to not have it STRICT?

**Review by a-thaler — CHANGES_REQUESTED** [review]

File: components/event-publisher-proxy/config/event-publisher-nats/120-peerauthentication.yaml (hunk 4 above)
Comment (jeremyharisch) [reply components/event-publisher-proxy/config/event-publisher-nats/120-peerauthentication.yaml:13]: Deleted it, it was not even used

File: resources/monitoring/values.yaml (hunk 2 above)
Comment (jeremyharisch) [reply resources/monitoring/values.yaml:702]: Those certs are not used. Already checked it and works as expected

File: components/event-publisher-proxy/config/event-publisher-nats/120-peerauthentication.yaml (hunk 4 above)
Comment (jeremyharisch) [reply components/event-publisher-proxy/config/event-publisher-nats/120-peerauthentication.yaml:13]: This PA is not used, will be removed

File: resources/monitoring/charts/kube-state-metrics/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -15,7 +15,8 @@ spec:
       {{- include "kube-state-metrics.selectorLabels" . | nindent 6 }}
   endpoints:
     - port: http
-      {{- if .Values.prometheus.monitor.honorLabels }}
+      scheme: http
+    {{- if .Values.prometheus.monitor.honorLabels }}
```
Comment (a-thaler) [inline resources/monitoring/charts/kube-state-metrics/templates/servicemonitor.yaml:19]: please revert the change :)

File: resources/monitoring/values.yaml [hunk 6]
```diff
@@ -681,13 +685,23 @@ grafana:
...
+      certFile: /etc/prometheus/secrets/istio.default/cert-chain.pem
+      keyFile: /etc/prometheus/secrets/istio.default/key.pem
+      insecureSkipVerify: true
+
 ## Component scraping the kube api server
 ##
 kubeApiServer:
   enabled: true
-  tlsConfig:
+
```
Comment (a-thaler) [inline resources/monitoring/values.yaml:699]: please revert all changes here

**Review by a-thaler — APPROVED** [review]

**Review by piotrmiskiewicz — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Review by sayanh — APPROVED** [review]

**Review by mjakobczyk — APPROVED** [review]
After checking it out and testing with Darek, it looks good to us 👍

**Comment by apenen** [comment]
Hi! Could you apply this change also in version 1.x? Thanks.
